sphinx_immaterial/external_resource_cache.py

```diff
@@ -81,14 +81,28 @@ def get_cache_dir(app: sphinx.application.Sphinx) -> str:
     return getattr(app.config, _RESOURCE_CONFIG_KEY)
 
 
+def _config_inited(
```

Comment:
This would probably be better in a separate file, e.g. `local_mathjax.py`, since it doesn't really relate to the rest of this file (and note that the google fonts are now handled by a separate file as well).

We also probably want a way to disable this behavior, so that users can still set their own mathjax path if desired.

Finally, it would be nice to only copy mathjax if actually used.  See this logic here:
https://github.com/sphinx-doc/sphinx/blob/4eb706eb21bc3aaa849f036593369ee1c503743d/sphinx/ext/mathjax.py#L82

Similarly for mermaid it would be nice to also only copy it if actually needed.

Reply:
> Similarly for mermaid it would be nice to also only copy it if actually needed.

I could make the mermaid extension optional, but you probably want a more context-specific solution (as suggested for mathjax.

Reply:
Yes that's what I was thinking.

Reply:
Well, I abstracted the config-inited code into its own module and removed the lines from `sphinx_immaterial/__init__.py` that copy the mathjax dist into the build dir. This seems to still work well: Mathjax script is still added to the HTML accordingly via `sphinx.ext.mathjax.install_mathjax()`.

I also added a condition to only modify the `mathjax_path` conf option if the value is set to the mathjax ext default path value. This tested well when I copied the mathjax dist into docs/_static/dist/mathjax.js & modified the conf.py accordingly.

---
However, I'm noticing new font issues spawn from the self-hosted mathjax script. From the web console:
```
downloadable font: download failed (font-family: "MJXZERO" style:normal weight:400 stretch:100 src index:0): status=2152857618 source: file:///path/to/sphinx-immaterial/docs/_build/html/_static/output/chtml/fonts/woff-v2/MathJax_Zero.woff
downloadable font: download failed (font-family: "MJXTEX" style:normal weight:400 stretch:100 src index:0): status=2152857618 source: file:///path/to/sphinx-immaterial/docs/_build/html/_static/output/chtml/fonts/woff-v2/MathJax_Main-Regular.woff
downloadable font: download failed (font-family: "MJXTEX-I" style:normal weight:400 stretch:100 src index:0): status=2152857618 source: file:///path/to/sphinx-immaterial/docs/_build/html/_static/output/chtml/fonts/woff-v2/MathJax_Math-Italic.woff
```

I think it means I need to copy these fonts from the mathjax npm pkg in node_modules; in which case it would probably be better to put the mathjax dist files in their own folder during npm build. Since the `sphinx.ext.mathjax` doesn't seem to handle the fonts, this seems to complicate things even further.

Reply:
Ok, I've got a working solution for mathjax (including the mathjax fonts now). I'll push it shortly.

---
The JS bundle already conditionally mounts the mermaid script if the page has a `<div class="mermaid">` in it. So, I should only need to devise a way to detect if any mermaid directives are invoked, then copy the mermaid dist when needed.

My initial guess is to use the builder env to persist a theme-specific flag for this, but maybe you have a better approach in mind?

Reply:
Yes, adding it in the env (and remember to handle merging in order to support parallel builds) is exactly what I had in mind.